jelares/portfolio/src/main/webapp/javascript/blogLogic.js

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+    Fetches the URL for uploading to Blobstore and adds it to the blog form
+*/
+function fetchBlobstoreUrl() {
+  fetch('/blobstore-upload-url')
+      .then((response) => {
+        return response.text();
+      })
+      .then((imageUploadUrl) => {
+        const messageForm = document.getElementById('blog-post');
+        messageForm.action = imageUploadUrl;
+      });
+}
+
```

Comment:
"Logic" is probably redundant in JS file names (since the files are presumed to include logic).

If you're struggling for a name, consider just "blog.js" if that's not too generic. You could also use "utilities" or "helpers", like "blog-helpers" or "blog-utilities", but these can be overused and should be avoided as much as possible.

Also note JS file naming convention:
https://google.github.io/styleguide/jsguide.html#file-name

Reply:
I'll be more careful in naming my files in the future - updated